dot/sync/worker_test.go

```diff
@@ -9,53 +9,84 @@ import (
 	"time"
 
 	"github.com/ChainSafe/gossamer/dot/network"
+	"github.com/ChainSafe/gossamer/lib/common/variadic"
 	"github.com/libp2p/go-libp2p/core/peer"
 	"github.com/stretchr/testify/require"
-	"go.uber.org/mock/gomock"
 )
 
 func TestWorker(t *testing.T) {
 	peerA := peer.ID("peerA")
-	ctrl := gomock.NewController(t)
-
-	reqMaker := NewMockRequestMaker(ctrl)
-	reqMaker.EXPECT().
-		Do(peerA, nil, gomock.AssignableToTypeOf((*network.BlockResponseMessage)(nil))).
-		DoAndReturn(func(_, _, _ any) any {
-			time.Sleep(2 * time.Second)
+	m := uint32(60)
+	blockReq := &network.BlockRequestMessage{
+		RequestedData: 1,
+		StartingBlock: variadic.Uint32OrHash{},
+		Direction:     3,
+		Max:           &m,
+	}
+	requestCounter := 0
+	reqMaker := fakeReqMaker{
```

Comment:
I don't think having a custom handmade `fakeReqMaker` is ok since we can use the generated mock (which is the approach in the project), so I would suggest to keep the `reqMaker` as it is